datashare-app/src/main/java/org/icij/datashare/utils/DocumentVerifier.java

```diff
@@ -39,7 +41,13 @@ public boolean isRootDocumentSizeAllowed(Document document) {
             return true;
         }
         long maxSizeBytes = getEmbeddedDocumentDownloadMaxSizeBytes();
-        Document rootDocument = indexer.get(document.getProjectId(), document.getRootDocument());
+        Document rootDocument;
+        Entity entity = indexer.get(document.getProjectId(), document.getRootDocument());
+        if (entity.getClass().equals(Duplicate.class)) {
+            rootDocument = indexer.get(document.getProjectId(), ((Duplicate) entity).documentId);
+        } else {
+            rootDocument = (Document) entity;
+        }
```

Comment:
This works but it would be cleaner to move it to a `getRootDocument` method that returns a Document instance.